File: Android/core/core-domain/src/main/java/com/youkong/core/domain/repository/FriendRepository.kt
package com.youkong.core.domain.repository

import com.youkong.core.domain.model.Friend
import com.youkong.core.domain.model.FriendRequest
import com.youkong.core.domain.model.SendFriendRequestResult
import kotlinx.coroutines.flow.Flow

interface FriendRepository {
    val friends: Flow<List<Friend>>
    val pendingRequestCount: Flow<Int>

    suspend fun getFriends(): Result<List<Friend>>

    suspend fun deleteFriend(userId: String): Result<Unit>

    suspend fun getInvitedByMe(): Result<List<Friend>>

    suspend fun getInvitedMe(): Result<List<Friend>>

    // 好友请求相关
    suspend fun sendFriendRequest(phone: String, message: String? = null): Result<SendFriendRequestResult>

    suspend fun getReceivedRequests(): Result<List<FriendRequest>>

    suspend fun getSentRequests(): Result<List<FriendRequest>>

    suspend fun handleFriendRequest(requestId: String, accept: Boolean): Result<Unit>

    suspend fun getPendingRequestCount(): Result<Int>
}

File: iOS/YouKong/Domain/Repositories/FriendRepository.swift
//
//  FriendRepository.swift
//  YouKong
//
//  好友系统 Repository 协议
//

import Foundation

protocol FriendRepositoryProtocol {
    /// 获取好友列表
    func getFriends() async throws -> [Friend]

    /// 删除好友
    func deleteFriend(userId: String) async throws

    /// 获取我邀请的好友
    func getFriendsInvitedByMe() async throws -> [InvitedFriend]

    /// 获取邀请我的好友
    func getFriendsWhoInvitedMe() async throws -> [InvitedFriend]
}
